gtsfm/bundle/two_step_ba.py

```diff
@@ -0,0 +1,97 @@
+"""2-step bundle adjustment with outlier filtering in the middle.
+
+Authors: Ayush Baid
+"""
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.common.pose_prior import PosePrior
+from gtsfm.bundle.bundle_adjustment import BundleAdjustmentOptimizer
+
+DEFAULT_INTERMEDIATE_REPROJ_ERROR_THRESH = 10
+FIRST_BA_NUM_ITERATIONS = 5
+
+
+class TwoStepBA(BundleAdjustmentOptimizer):
+    """Bundle adjustment optimizer which performs bundle adjustment twice: runs for a few iterations, filters out the
+    outlier tracks, and then runs for the remaining iterations with the inlier tracks."""
+
+    def __init__(
+        self,
+        intermediate_reproj_error_thresh: float = DEFAULT_INTERMEDIATE_REPROJ_ERROR_THRESH,
+        output_reproj_error_thresh: Optional[float] = None,
+        robust_measurement_noise: bool = False,
+        shared_calib: bool = False,
+        max_iterations: Optional[int] = None,
+    ) -> None:
+        """Initialize the optimizer.
+
+        Args:
+            intermediate_reproj_error_thresh (optional): The reprojection error threshold used to filter out tracks
```

Comment:
Thank for the PR, Ayush. Could you do a hyperparameter search to check if we need separate intermediate and final outlier thresholds, or if one would suffice?